lib/puppet/provider/user/windows_adsi.rb

```diff
@@ -123,16 +307,82 @@ def password
     user.password_is?( @resource[:password] ) ? @resource[:password] : nil
   end
 
+  # Sets the user's password.
+  #
+  # @api private
+  def set_password(password, existing_user)
+    if existing_user
+      # We always want to set a new user's password. However for
+      # an existing user, we need to check if changing their
+      # password makes sense
+
+      # Do NOT attempt to retrieve a new user's attributes. Doing so
+      # will raise an exception because the user's ADSI property values
+      # (some of which correspond to their attributes) are only loaded
+      # into the property cache _after_ they've been committed.
+      current_attributes = attributes
+
+      if current_attributes[:password_change_required] == 'true'
```

Comment:
Does this make sense? We did not previously print a warning when `password_change_required` was set.